flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -150,21 +156,23 @@ public ResourceProfile(
 			extendedResources);
 	}
 
-	@VisibleForTesting
-	public ResourceProfile(double cpuCores, MemorySize taskHeapMemory) {
```

Comment:
These constructors for combinations of different given resources and different raw/advanced types is not flexible. It's also not clear to track the resources that are set.
So I think it's better to introduce a builder for ResourceProfile, as proposed in FLINK-14733.